src/main/java/edu/ucla/library/libcal/handlers/ProxyHandler.java

```diff
@@ -0,0 +1,121 @@
+import edu.ucla.library.libcal.services.LibCalProxyService;
+import edu.ucla.library.libcal.services.OAuthTokenService;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A handler that processes status information requests.
+ */
+@SuppressWarnings("PMD.AvoidCatchingGenericException")
+public class ProxyHandler implements Handler<RoutingContext> {
+
+    /**
+     * The handler's logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProxyHandler.class, MessageCodes.BUNDLE);
+
+    /**
+     * A constant for evaluating request path.
+     */
+    private static final String LIBCAL = "libcal";
+
+    /**
+     * The handler's copy of the Vert.x instance.
+     */
+    private final Vertx myVertx;
+
+    /**
+     * A service for LibCal API calls
+     */
+    private final LibCalProxyService myApiProxy;
+
+    /**
+     * A service for LibCal OAuth calls
+     */
+    private final OAuthTokenService myTokenProxy;
+
+    /**
+     * Creates a handler that returns a status response.
+     *
+     * @param aVertx A Vert.x instance
+     */
+    public ProxyHandler(final Vertx aVertx) {
+        myVertx = aVertx;
+        myApiProxy = LibCalProxyService.createProxy(myVertx);
+        myTokenProxy = OAuthTokenService.createProxy(myVertx);
+    }
+
+    @Override
+    public void handle(final RoutingContext aContext) {
+        final HttpServerResponse response = aContext.response();
+        final String path = aContext.request().path();
+
+        try {
+            if (path.endsWith(LIBCAL) || path.endsWith(LIBCAL.concat(SLASH))) {
+                response.setStatusCode(HTTP.BAD_REQUEST).putHeader(HttpHeaders.CONTENT_TYPE, TEXT_PLAIN.toString())
+                        .end(LOGGER.getMessage(MessageCodes.LCP_006));
+            } else {
+                final String receivedQuery = path.replaceAll(LIBCAL.concat(SLASH), "")
+                        .concat(aContext.request().query() != null ? aContext.request().query() : "");
+                myTokenProxy.getBearerToken().compose(token -> {
+                    return myApiProxy.getLibCalOutput(token, SLASH.concat(receivedQuery)).onSuccess(apiOutput -> {
+                        response.setStatusCode(HTTP.OK).putHeader(HttpHeaders.CONTENT_TYPE,
+                                APPLICATION_JSON.toString());
+                        response.end(apiOutput);
+                    });
+                }).onFailure(failure -> {
+                    returnError(response, HTTP.INTERNAL_SERVER_ERROR, failure.getMessage());
+                });
+            }
+        } catch (Exception details) {
+            returnError(response, HTTP.INTERNAL_SERVER_ERROR, details.getMessage());
+        }
```

Comment:
I would advocate for removing this try-catch, since it discourages the writing of a test suite that would discover things like accidental null references.

Reply:
Removed try-catch